streaming/stream_transfer_task.cc

```diff
@@ -203,15 +203,27 @@ future<> send_mutation_fragments(lw_shared_ptr<send_info> si) {
         }();
 
         auto sink_op = [sink, si, got_error_from_peer] () mutable -> future<> {
-            return do_with(std::move(sink), [si, got_error_from_peer] (rpc::sink<frozen_mutation_fragment, stream_mutation_fragments_cmd>& sink) {
-                return repeat([&sink, si, got_error_from_peer] () mutable {
-                    return si->reader(db::no_timeout).then([&sink, si, s = si->reader.schema(), got_error_from_peer] (mutation_fragment_opt mf) mutable {
-                        if (mf && !(*got_error_from_peer)) {
+            bool partition_opened = false;
+            return do_with(std::move(sink), partition_opened, [si, got_error_from_peer] (rpc::sink<frozen_mutation_fragment, stream_mutation_fragments_cmd>& sink, bool& partition_opened) {
```

Comment:
You can just replace `partition_opened` here with `false` and then you don't have to declare it above.

Reply:
Yes we can, but I intentionally declare the variable to give the false more context.

Reply:
Ok, fair enough, I will not complain about more context.